molgenis-data/src/main/java/org/molgenis/data/listeners/EntityListenersService.java

```diff
@@ -0,0 +1,43 @@
+package org.molgenis.data.listeners;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.SetMultimap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PreDestroy;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Service
+public class EntityListenersService
+{
+	private static final ConcurrentHashMap<String, SetMultimap<Object, EntityListener>> entityListeners = new ConcurrentHashMap<String, SetMultimap<Object, EntityListener>>();
+
+	/**
+	 * returns a SetMultimap<Object, EntityListener> where the object is the id of the entity and the EntityListener is listener interface
+	 *
+	 * @param repoFullName
+	 * @return SetMultimap<Object, EntityListener>
+	 */
+	public synchronized SetMultimap<Object, EntityListener> getEntityListeners(String repoFullName)
```

Comment:
This one probably shouldn't be synchronized cause it'll get executed a LOT of times concurrently.